obearon/discord/events/assign_hibernation.py

```diff
@@ -0,0 +1,45 @@
+"""
+Assign hibernation to users out of clan task.
+"""
+
+from datetime import datetime
+from datetime import timedelta
+from datetime import timezone
+
+from discord import Forbidden
+from loguru import logger
+
+from obearon.database import crud
+
+
+async def on_application_command_completion(ctx) -> None:
+    """
+    Assign hibernation role to users not in clan that have verified role.
+
+    Args:
+        ctx: Context of completed commands.
+    """
+    if ctx.command.name == "upload_clan_inventory":
+        guild_id = ctx.guild.id
+        guild = ctx.guild
+        verified_role = await crud.get_verify_role(guild_id)
+        hibernation_role = await crud.get_hibernation_role(guild_id)
+        warframe_players = await crud.get_warframe_player_names_not_in_clan()
+
+        for member in guild.members:
+            if not member.get_role(verified_role.verified_role_id):
+                continue
+            joined_recently = datetime.now(timezone.utc) - member.joined_at < timedelta(hours=24)
+            if joined_recently:
+                continue
+            if not any(name.lower() == member.nick.lower() for name in warframe_players):
```

Comment:
Another note for you that you can just ignore if you want to, `.lower()` for matching strings is only okay because we know that warframe names are `[a-z0-9_.-]`. For more complex cases, you'd want `casefold()`.